### PR TITLE
Refactor compile_gpdb.bash

### PR DESCRIPTION
**Pull request by bmdoil**
Build environment setup tasks removed from compile_gpdb.bash.
Build images now fully resolve build dependencies.

Authored-By: Brent Doil <bdoil@vmware.com>

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [x] Document changes
- [x] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: concourse/pipelines/gpdb_6X_STABLE-generated.yml [hunk 1]
```diff
@@ -2927,7 +2739,7 @@ jobs:
               vars:
                 <<: *ccp_default_vars
                 instance_type: n1-standard-2
-                
+                number_of_nodes: 3
```
Comment (khuddlefish) [inline concourse/pipelines/gpdb_6X_STABLE-generated.yml:2742]: where is this value coming from?

File: concourse/pipelines/gpdb_6X_STABLE-generated.yml [hunk 2]
```diff
@@ -1062,21 +924,11 @@ jobs:
...
-      - get: libsigar-installer
-        resource: libsigar-centos6
-      - get: python-tarball
-        resource: python-centos6
   - task: compile_gpdb
     file: gpdb_src/concourse/tasks/compile_gpdb.yml
     image: gpdb6-centos6-build
     params:
       CONFIGURE_FLAGS: ((configure_flags_with_extensions))
-      TARGET_OS: centos
```
Comment (khuddlefish) [inline concourse/pipelines/gpdb_6X_STABLE-generated.yml:1078]: do we derive the os and version from the image, or just don't need it?

File: concourse/pipelines/gpdb_6X_STABLE-generated.yml [hunk 3]
```diff
@@ -1086,7 +938,7 @@ jobs:
           file: gpdb_artifacts/bin_gpdb.tar.gz
       - put: bin_gpdb_clients_centos6
         params:
-          file: gpdb_artifacts/gpdb-clients-centos6.tar.gz
```
Comment (khuddlefish) [inline concourse/pipelines/gpdb_6X_STABLE-generated.yml:1089]: is there any impact from removing the os version from the filename?

File: concourse/pipelines/gpdb_6X_STABLE-generated.yml (hunk 1 above)
Comment (bmdoil) [reply concourse/pipelines/gpdb_6X_STABLE-generated.yml:2742]: https://github.com/greenplum-db/gpdb/commit/2ccd0c8510c6344f74715a1ccedcd4ec0d914e88

The regenerated prod pipeline wasn't committed, so it's appearing as a diff here.

File: concourse/pipelines/gpdb_6X_STABLE-generated.yml (hunk 2 above)
Comment (bmdoil) [reply concourse/pipelines/gpdb_6X_STABLE-generated.yml:1078]: It's derived from the image.

File: concourse/pipelines/gpdb_6X_STABLE-generated.yml (hunk 3 above)
Comment (bmdoil) [reply concourse/pipelines/gpdb_6X_STABLE-generated.yml:1089]: No, it's tech debt. We rename it anyway when we upload to blob store.

**Review by 50wu — APPROVED** [review]
Looks good to me

File: concourse/pipelines/gpdb_6X_STABLE-generated.yml (hunk 1 above)
Comment (kalensk) [reply concourse/pipelines/gpdb_6X_STABLE-generated.yml:2742]: Adding support to what Brent mentioned.

Comment (khuddlefish) [reply concourse/pipelines/gpdb_6X_STABLE-generated.yml:2742]: ok cool. thanks for the info

**Review by khuddlefish — APPROVED** [review]

**Pull request merged**